### ISSUE TITLE
Is there a negative performance impact from using Typescript with k6 over JS? If so, what is it?

### ISSUE THREAD

**Issue by nicholaspshaw** [grafana/k6-template-typescript#47]
Covered everything in the title.  I started using your template to migrate from JS to TS but I'm seeing some articles claiming there's a negative performance hit such as : https://medium.com/@mohsenny/setting-up-a-robust-performance-and-load-testing-framework-with-k6-bfbaadd39b41

I'm curious if this is just hearsay or has been measured in any meaningful way

**Comment by allansson** [comment]
> k6 is designed to run JavaScript files, not TypeScript or other languages. This constraint is aligned with k6’s goal of streamlined performance testing, as JavaScript is universally compatible and generally more efficient for the engine k6 uses.

Are you referring to this section?

What this repository does is that it provides you with a build step to convert typescript into javascript. The process of building a bundle will not affect runtime performance in any way.

And even if k6 had native support for typescript, there would be no change in runtime performance. The script would be parsed and all types would be thrown away, essentially turning it into JavaScript. This is what other runtimes, e.g. bun and deno, do.

TLDR; using typescript will not hurt performance
